src/canvas/graphic.ts

```diff
@@ -151,15 +152,23 @@ function brushPath(ctx: CanvasRenderingContext2D, el: Path, inBatch: boolean) {
         }
     }
 
-    const lineDash = style.lineDash;
-    const lineDashOffset = style.lineDashOffset;
+    let lineDash = style.lineDash;
+    let lineDashOffset = style.lineDashOffset;
 
     const ctxLineDash = !!ctx.setLineDash;
 
     // Update path sx, sy
     const scale = el.getGlobalScale();
     path.setScale(scale[0], scale[1], el.segmentIgnoreThreshold);
 
+    if (lineDash) {
+        const lineScale = (style.strokeNoScale && el.getLineScale) ? el.getLineScale() : 1;
+        lineDash = map(lineDash, function (rawVal) {
+            return rawVal / lineScale;
```

Comment:
It's better to only been mapped when `lineScale` is not `1` to reduce the potential cost.